main.py

```diff
@@ -0,0 +1,82 @@
+import time
+import argparse
+from os.path import join, dirname
+
+from common import (get_dict_from_yaml, JUMP_HOST_CONFIG_PATH, JUMP_HOST_DIR,
+                    JUMP_HOST_LICENSE_PATH, JUMP_HOST_SSH_KEY_PATH,
+                    OLD_JUMP_HOST_DIR, logger, VM)
+
+INSTALL_CLUSTER_SCRIPT = 'install_cluster.py'
+
+
+def _prepare_jump_host(jump_host, license_path, config_path):
+    commands_list = [
+        'sudo yum install -y epel-release',
+        'sudo yum install -y python-pip',
+        'sudo yum groupinstall -y \"Development Tools\"',
+        'sudo yum install -y python-devel',
+        'sudo pip install --upgrade pip==9.0.1',
+        'sudo pip install -r {0}'.format(join(JUMP_HOST_DIR,
+                                              'requirements.txt'))
+    ]
+    scp_files_list = [
+        'requirements.txt',
+        INSTALL_CLUSTER_SCRIPT,
+        'common.py',
+        'manager_config.yaml',
+        'postgresql_config.yaml',
+        'rabbitmq_config.yaml',
+    ]
+    logger.info('Preparing the jump-host')
+    if jump_host.path_exists(JUMP_HOST_DIR):
+        jump_host.run_command('rm -rf {}'.format(OLD_JUMP_HOST_DIR))
+        jump_host.run_command('mv {0} {1}'.format(JUMP_HOST_DIR,
+                                                  OLD_JUMP_HOST_DIR))
+    jump_host.run_command('mkdir {0}'.format(JUMP_HOST_DIR))
+
+    for file_name in scp_files_list:
+        jump_host.put_file(join(dirname(__file__), file_name),
+                           join(JUMP_HOST_DIR, file_name))
+    for command in commands_list:
+        jump_host.run_command(command)
+
+    jump_host.put_file(jump_host.key_file_path, JUMP_HOST_SSH_KEY_PATH)
```

Comment:
If we did scrap the jump host approach (and require this to instead be run on a jump host) then we'd be able to get rid of a lot of code here, and drop it down to just doing an ssh check on the target nodes I think?